spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/storage/queue/StorageQueueCheckpointer.java

```diff
@@ -26,7 +27,13 @@ public StorageQueueCheckpointer(CloudQueue cloudQueue) {
 
     @Override
     public CompletableFuture<Void> checkpoint(CloudQueueMessage cloudQueueMessage) {
-        // Storage queue unsupported
-        throw new UnsupportedOperationException();
+        CompletableFuture<Void> completableFuture = CompletableFuture.runAsync(() -> {
```

Comment:
If you just run it in another thread, I don't think the exception will be thrown by future.